### PR TITLE
Document

### PR DESCRIPTION
**Pull request by TangyrenMidKing**
All existing event endpoints are documented. 
- Post /create_event
- Delete /delete_event/{event_id}
- Get /get_event/{event_id}
- Get /get_next_event_page
- Get /get_recent_events
- Post /update_event

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Firebase credentials
 key.json
+.firebaserc
```
Comment (Astatium5) [inline .gitignore:3]: No need to add these files to .gitignore. They don't contain any sensitive info and needed to run the Firebase Emulator Suite

**Review by Astatium5 — CHANGES_REQUESTED** [review]

File: src/__main__.py [hunk 2]
```diff
@@ -64,4 +64,4 @@ def page_not_found(e) -> Response:
 
 
 if __name__ == '__main__':
-    app.run(host='0.0.0.0', port=int(environ.get("PORT", 8080)), debug=True)
```
[the remark this answers is not in the document]
Comment (TangyrenMidKing) [reply src/__main__.py:67]: oh, my bad, for get change it, it only for my local test.

File: src/schemas.yml [hunk 3]
```diff
@@ -31,3 +15,38 @@ EventData:
...
+
+AssignRole:
+  type: object
+  properties:
+    email:
+      type: string 
+      example: admin@gmail.com
+    role: 
+      type: string
+      example: Doctor
```
Comment (Astatium5) [inline src/schemas.yml:52]: Add a blank line at the end of the file

File: src/api/roles.py [hunk 4]
```diff
@@ -91,26 +122,44 @@ def create_role() -> Response:
...
+            type: string
+          required: true
+    responses:
+        200:
+            content:
+                application/json:
+                    schema:
+                        type: array
+                        items:
+                            type: string
```
Comment (Astatium5) [inline src/api/roles.py:152]: No response schema here

File: src/api/roles.py [hunk 5]
```diff
@@ -27,10 +27,22 @@
...
+        Granted a role for newly registering user.
+    ---
+    tags:
+        - role
+    summary: Upon newly registering a user, they are granted a role if their email used is matched to one in Roles/roleList document.
+    parameters:
+        - in: header
+          name: Authorization
+          schema:
+            type: string
```
Comment (Astatium5) [inline src/api/roles.py:39]: Add an example here

File: src/api/events.py [hunk 6]
```diff
@@ -184,10 +270,30 @@ def get_recent_events() -> Response:
...
+        - in: header
+          name: ID
+          schema:
+            type: string
+          required: true
+    responses:
+        200:
+            content:
+                application/json:
+                    schema:
```
Comment (Astatium5) [inline src/api/events.py:293]: No response schema here

**Review by Astatium5 — CHANGES_REQUESTED** [review]

**Review by Astatium5 — APPROVED** [review]

**Pull request merged**